sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/V2ExpressionUtils.scala

```diff
@@ -205,4 +206,171 @@ object V2ExpressionUtils extends SQLConfHelper with Logging {
         None
     }
   }
+
+  def toCatalyst(expr: V2Expression): Option[Expression] = expr match {
```

Comment:
@cloud-fan, I went for a simpler option and added the conversion to `V2ExpressionUtils`. I initially thought about a single utility that could do bi-directional conversion but it probably won't be worth the complexity and the risk of introducing bugs in such a critical path.